### PR TITLE
Implementing open()/close() on HappyBase connection.

### PR DESCRIPTION
**Pull request by dhermes**
Also using `open()` in constructor and `close()` in destructor, as is done in the thrift version of HappyBase.

### CODE REVIEW

File: gcloud/bigtable/happybase/connection.py [hunk 1]
```diff
@@ -174,3 +178,30 @@ def _handle_legacy_args(arguments_dict):
...
+        """Close the underlying transport to Cloud Bigtable.
+
+        This method closes the underlying HTTP/2 gRPC connection using a
+        :class:`.Client` bound to the :class:`.Cluster` owned by
+        this connection.
+        """
+        self._cluster._client.stop()
+
+    def __del__(self):
+        try:
```
Comment (tseaver) [inline gcloud/bigtable/happybase/connection.py]: Setting a class default of `_initialized = False` would make this method pellucid:

```python
    def __del__(self):
        if self._initialized:
            self.close()
```

Comment (dhermes) [inline gcloud/bigtable/happybase/connection.py]: :+1: I think I originally just copied what the HappyBase folks did

Comment (dhermes) [inline gcloud/bigtable/happybase/connection.py]: Also, I have **never** heard `pellucid` until just now

Comment (tseaver) [inline gcloud/bigtable/happybase/connection.py]: Actually, isn't the `_initialized` flag redundant, given the `_cluster` attribute?  If it was set to `None` at the class level, then you could make `__del__` as:

```python
    def __del__(self):
        if self._cluster is not None:
            self.close()
```

**Comment by dhermes** [comment]
@tseaver PTAL

**Comment by tseaver** [comment]
LGTM

**Comment by dhermes** [comment]
I'm unclear why AppVeyor says it fails. It seems the relevant build is passing:
https://ci.appveyor.com/project/dhermes/gcloud-python/build/1.0.150.master

**Pull request merged**